src/software/ai/evaluation/calc_best_shot.cpp

```diff
@@ -27,31 +14,22 @@ std::optional<Shot> calcBestShotOnGoal(const Point &goal_post_neg,
     {
         obs.push_back(Circle(robot.position(), ROBOT_MAX_RADIUS_METERS));
     }
-    return calcMostOpenDirectionFromCircleObstacles(
-        p, Segment(goal_post_neg, goal_post_pos), obs);
-}
-std::optional<Shot> calcBestShotOnGoal(const Point &goal_post_neg,
-                                       const Point &goal_post_pos, const Point &p,
-                                       const std::vector<Circle> &obstacles)
-{
-    // Use shot evaluation function to get the best Shot
-    return calcMostOpenDirectionFromCircleObstacles(
-        p, Segment(goal_post_neg, goal_post_pos), obstacles);
+    return calcMostOpenDirectionFromCircleObstacles(shot_origin, goal_post, obs);
 }
 
 std::optional<Shot> calcBestShotOnGoal(const Field &field, const Team &friendly_team,
-                                       const Team &enemy_team, const Point &point,
-                                       bool shoot_on_enemy_goal, double radius,
+                                       const Team &enemy_team, const Point &shot_origin,
+                                       TeamType goal, double radius,
                                        const std::vector<Robot> &robots_to_ignore)
 {
-    std::vector<Circle> obstacles;
+    std::vector<Robot> obstacles;
```

Comment:
is there a reason why we're keeping `calcBestShotOnGoal` with `const std::vector<Robot> &obstacles` instead of `const std::vector<Circle> &obstacles`?

Reply:
IIRC all the usages were with `<Robot>` in the file except for one, which I modified to use `Robot` instead. Otherwise, there is no particular reason. I just thought the common way to think about it is you would be calling it with robots on the field, not circles.